docs/lesson-plans/week-5.md

```diff
@@ -0,0 +1,569 @@
+# Week 5: Advanced Backend and Testing
+
+## Agenda
+- Backend Basics Cont.
+- Testing
+  - JUnit 5
+  - Mockito
+- Advanced Backend
+  - DTOs in General
+  - Unmarshalling objects into DTOs
+  - Backend Validation
+  - Other Request Types
+  
+## Basic Plan
+1. Talk about testing in the software development world
+  - Types of tests
+  - Basic testing with JUnit
+    - Show basic test examples
+    - `BeforeEach`/`BeforeAll` annotations
+    - Parameterized tests
+    - `Externals` Class
+    - Exceptions
+  - Mocking objects with Mockito
+    - Mocking behavior and returns
+    - Verification
+    - Argument Captors
+    - Show the test created for the backend api module
+    
+2. DTOs and Marshalling/Unmarshalling
+  - What are DTOs
+    - What are they used for
+  - Marshalling/Unmarshalling
+  - Validating unmarshalled objects/why it's necessary
+ 
+3. New Request Types
+  - `POST`, `PUT`, `PATCH`, and `DELETE`: what are they commonly used for
+  
+## Testing
+
+### Types of Tests
+
+As I'm sure you can tell from the name, testing is the method used to help prove that your code works the way you expect 
```

Comment:
maybe switch the wording from "As I'm sure..." to "You may be able to tell..."

Reply:
👍 fixed